src/vaadin-date-time-picker-custom-field.html

```diff
@@ -26,15 +26,20 @@
         return 'vaadin-date-time-picker-custom-field';
       }
 
-      checkValidity() {
-        const invalidFields = this.inputs
-          .filter(input => !(input.checkValidity || input.validate).call(input));
+      connectedCallback() {
+        this.__toggleHasValue = function(value) {
+          if (value !== null && value.trim() !== '' && value !== 'T') {
+            this.setAttribute('has-value', '');
+          } else {
+            this.removeAttribute('has-value');
+          }
```

Comment:
I noticed that this now results in `has-value` being set also when only one of the two fields has a value (because then custom-field value is e.g. `"2020-01-17T"` or `"T12:00"`).

Using this instead seems to work better:
```js
if (value !== null && value !== '' && value.split('T').indexOf('') === -1) {
```
(I don't think there's a need for `trim()` since the values should not have spaces anyway)

Reply:
With my suggestion the required indicator is shown also in case when one of the fields is empty (and DTP has no value) which is slightly better imo even though in this case the custom field technically does have a value but I think this behaviour is still more logical in context of DTP and custom field is just an implementation detail in this case.